### PR TITLE
mhs/CUMULUS-2492/handle-trailing-underscores-in-shortname

### PR DESCRIPTION
**Pull request by flamingbear**
**Summary:**

Ports deconstructCollectionId from @cumulus/message/Collections and uses that to ensure collections with trailing short names are properly encoded and decoded into collectionIds

Addresses [CUMULUS-2492: As a GESDISC Cumulus Integrator, I'd like cumulus to handle collections names with underscores such as "S5P_L2__CO____"](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2492)

## Changes

* Simple upgrade of package-lock.json
* Adds updated deconstructCollectionId function
* Modifies existing code to use deconstructCollectionId

## Testing

As part of testing, I created a collection and ingested data that was posted to
CMR the collection shortname was `trailing____` with a version of `1.0` You can
see the successful exectutions in the SBX stepfunctions
mhs3-IngestAndPublishGranule:2a3e000f-c523-4fd6-bd20-4155b1b99515

Successfully created/deleted collections with trailing underscores in the short names on the dashboard.

## Core requirements
There is a corresponding CORE PR that is needed for this to work in practice https://github.com/nasa/cumulus/pull/2723
I won't put an integration tests that has this requirement to keep from having to release an alpha of cumulus.  🥊  I don't think it requires one.

## PR Checklist

- [X] Update CHANGELOG
- [X] Unit tests
- [X] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

File: app/src/js/components/DeleteCollection/DeleteCollectionModal.js [hunk 1]
```diff
@@ -40,7 +37,7 @@ class DeleteCollectionModal extends React.Component {
         <p> You have submitted a request to delete the following collection: </p>
         <br />
         <h1>
-          <strong>{collectionLabelForId(this.props.collectionLabel)}</strong>
+          <strong>${this.props.collectionLabel}</strong>
```
Comment (kkelly51) [inline app/src/js/components/DeleteCollection/DeleteCollectionModal.js:40]: So I see this extra "$" in the prompt.

Comment (kkelly51) [reply app/src/js/components/DeleteCollection/DeleteCollectionModal.js:40]: Which only shows up on the single collection page when you go to delete. 
Not on the collections list page when you can select multiple to delete at a time.

**Review by kkelly51 — APPROVED** [review]
1 little comment

File: app/src/js/components/DeleteCollection/DeleteCollectionModal.js (hunk 1 above)
Comment (flamingbear) [reply app/src/js/components/DeleteCollection/DeleteCollectionModal.js]: Oh, yes, that's bad.  https://github.com/nasa/cumulus-dashboard/pull/1051/commits/b2a1d5799dfed34f633332a98f70b568b5a06199

**Pull request merged**